pages/docs/learn/durable-execution.mdx

```diff
@@ -0,0 +1,99 @@
+
+In comparison, some durable execution systems modify the runtime environment to persist state or interrupt errors or exceptions. Inngest functions are defined with steps written in normal language code, which enables them to run in any environment or runtime - including serverless environments - without modification.
+
+### How steps are executed
+
+Inngest functions are defined with a series of steps that define the execution flow of the function. Each step is defined with a unique ID and a function that defines the work to be done. The data returned can be used by subsequent steps.
+
+Inngest functions are executed incrementally, _step by step_. As functions are executed, the results of each step are returned to Inngest and persisted in a managed function state store. The function is then re-executed and any steps that have already been successfully executed are [_memoized_](https://en.wikipedia.org/wiki/Memoization). This means that the successfully executed steps are skipped and the SDK injects the data returned by the previous execution into the function.
+
+Let's look at an example of a function and walk through how it is executed:
+
+```typescript
+const fn = inngest.createFunction(
+  { id: "import-contacts" },
+  { event: "contacts/csv.uploaded" },
+  // The function handler:
+  async ({ event, step }) => {
+    const rows = await step.run("parse-csv", async () => {
+      return await parseCsv(event.data.fileURI);
+    });
+
+    const normalizedRows = await step.run("normalize-raw-csv", async () => {
+      const normalizedColumnMapping = getNormalizedColumnNames();
+      return normalizeRows(rows, normalizedColumnMapping);
+    });
+
+    const results = await step.run("input-contacts", async () => {
+      return await importContacts(normalizedRows);
+    });
+
+    return { results };
+  }
+```
+
+### Initial execution
+
+1. When the function is first called, the _function handler_ is called with only the `event` payload data sent.
+2. When the first step is discovered, the `"parse-csv"` step is run. As the step has not been executed before, the step's code (the callback function) is run and the result is captured.
+3. The function does not continue executing beyond this step. Each SDK uses a different method to interrupt the function execution before running any more code in your function handler.
+4. Internally, the step's id (`"parse-csv"`) is hashed as the state identifier to be used in future executions. Additionally, the steps' index (`0` in this case) is also included in the result.
+5. The result is sent back to Inngest and persisted in the function state store.
+
+### Secondary executions - Memoization of steps
+
+Steps executed after the first step leverage the state of previous executions and memoization. Here's how it works:
```

Comment:
```suggestion
Each of the subsequent steps leverages the state of previous executions and memoization. Here's how it works:
```

I don't know if this is a good rephrasing. The original sentence sounds a bit confusing.